src/core_plugins/tagcloud/public/tag_cloud_controller.js

```diff
@@ -1,24 +1,20 @@
-import _ from 'lodash';
 import { uiModules } from 'ui/modules';
 import TagCloud from 'plugins/tagcloud/tag_cloud';
 import AggConfigResult from 'ui/vis/agg_config_result';
-import { FilterBarClickHandlerProvider } from 'ui/filter_bar/filter_bar_click_handler';
 
 const module = uiModules.get('kibana/tagcloud', ['kibana']);
-module.controller('KbnTagCloudController', function ($scope, $element, Private, getAppState) {
+module.controller('KbnTagCloudController', function ($scope, $element) {
 
   const containerNode = $element[0];
-  const filterBarClickHandler = Private(FilterBarClickHandlerProvider);
   const maxTagCount = 200;
   let truncated = false;
 
   const tagCloud = new TagCloud(containerNode);
   tagCloud.on('select', (event) => {
-    const appState = getAppState();
-    const clickHandler = filterBarClickHandler(appState);
     const aggs = $scope.vis.getAggConfig().getResponseAggs();
+    if (!aggs) return;
     const aggConfigResult = new AggConfigResult(aggs[0], false, event, event);
-    clickHandler({ point: { aggConfigResult: aggConfigResult } });
+    $scope.vis.API.events.filter({ point: { aggConfigResult: aggConfigResult } });
```

Comment:
this is still messy .... 

ideally filter event would take the aggConfig and value (instead of using aggConfigResult)
will address this in separate PR